packages/calypso-e2e/src/lib/pages/stats-page.ts

```diff
@@ -0,0 +1,41 @@
+import { Page } from 'playwright';
+
+export type StatsTabs = 'Traffic' | 'Insights' | 'Store';
+
+const selectors = {
+	tabItems: ( name: StatsTabs ) => `a:has-text("${ name }")`,
+};
+
+/**
+ * Represents the Statistics page.
+ */
+export class StatsPage {
+	private page: Page;
+
+	/**
+	 * Constructs an instance of the page.
+	 *
+	 * @param {Page} page Underlying page on which interactions take place.
+	 */
+	constructor( page: Page ) {
+		this.page = page;
+	}
+
+	/**
+	 * Given a string, click on the tab name on the page.
+	 *
+	 * @param {StatsTabs} name Name of the tab to click.
+	 * @returns {Promise<void>} No return value.
+	 */
+	async clickTabItem( name: StatsTabs ): Promise< void > {
+		await this.page.click( selectors.tabItems( name ) );
+
+		// Confirm the tab is now selected, by matching the `li` that holds the
+		// selector, then checking whether the `li` element has a class
+		// `is-selected`.
+		await this.page.waitForFunction(
```

Comment:
Optional: but since you're waiting for a very specific state in the DOM, why not just use `waitForSelector` here? I think it would read a little cleaner than the `waitForFunction` implementation.

E.g.
```typescript
await this.page.waitForSelector( `*css=li.is-selected >> ${ selectors.tabItems( name ) }`);
```

Reply:
Good question. No particular reason for choosing `waitForFunction`, just that it has prior use case in the codebase eg. [PricingTableBlock](https://github.com/Automattic/wp-calypso/blob/db66a1316b2820f4130c2d57cb035fdc0e927b99/packages/calypso-e2e/src/lib/blocks/pricing-table-block.ts#L60), [ThemesPage](https://github.com/Automattic/wp-calypso/blob/3da8bc6ebc2724b8bc06df6723ed98e73a3f41ac/packages/calypso-e2e/src/lib/pages/themes-page.ts#L64).

It seems that a simple `waitForFunction` might work here, so I can switch it to that (and makes sure tests do continue to pass).

Reply:
`waitForFunction` works just as well, thanks for the suggestion!